File: practice/algorithms/bit_manipulation/maximizing_xor/maximizing_xor.rb
def maxXor(l, r)
  max = 0
  for i in l..r do
    for j in l..r do
      xor = i ^ j
      max = xor if xor > max
    end
  end
  return max
end


l = gets.to_i
r = gets.to_i
print maxXor(l, r)

File: practice/algorithms/bit_manipulation/maximizing_xor/MaximizingXor.scala
import scala.io.Source

object MaximizingXor extends App {
  val lines = Source.stdin.getLines().toList
  val l = lines(0).toInt
  val r = lines(1).toInt
  val maxXor = (l: Int, r: Int) => {
    var max = 0
    (l to r).foreach(
      i => {
        (l to r).foreach(
          j => {
            val xor = i ^ j
            if (xor > max) max = xor
          })
      }
    )
    max
  }

  println(maxXor(l, r))
}
